src/kagglehub/gcs_upload.py

```diff
@@ -138,51 +148,111 @@ def _upload_blob(file_path: str, model_type: str) -> str:
     return response["token"]
 
 
-def zip_file(args: Tuple[Path, Path, Path]) -> int:
-    file_path, zip_path, source_path_obj = args
-    arcname = file_path.relative_to(source_path_obj)
-    size = file_path.stat().st_size
-    with zipfile.ZipFile(zip_path, "a", zipfile.ZIP_STORED, allowZip64=True) as zipf:
-        zipf.write(file_path, arcname)
-    return size
-
-
-def zip_files(source_path_obj: Path, zip_path: Path) -> List[int]:
-    files = [file for file in source_path_obj.rglob("*") if file.is_file()]
-    args = [(file, zip_path, source_path_obj) for file in files]
-
-    with Pool() as pool:
-        sizes = pool.map(zip_file, args)
-    return sizes
-
-
-def upload_files(source_path: str, model_type: str) -> List[str]:
-    source_path_obj = Path(source_path)
-    with TemporaryDirectory() as temp_dir:
-        temp_dir_path = Path(temp_dir)
-        total_size = 0
-
-        if source_path_obj.is_dir():
-            for file_path in source_path_obj.rglob("*"):
-                if file_path.is_file():
-                    total_size += file_path.stat().st_size
-        elif source_path_obj.is_file():
-            total_size = source_path_obj.stat().st_size
-        else:
-            path_error_message = "The source path does not point to a valid file or directory."
-            raise ValueError(path_error_message)
-
-        with tqdm(total=total_size, desc="Zipping", unit="B", unit_scale=True, unit_divisor=1024) as pbar:
-            if source_path_obj.is_dir():
-                zip_path = temp_dir_path / "archive.zip"
-                sizes = zip_files(source_path_obj, zip_path)
-                for size in sizes:
-                    pbar.update(size)
-                upload_path = str(zip_path)
-            elif source_path_obj.is_file():
-                temp_file_path = temp_dir_path / source_path_obj.name
-                shutil.copy(source_path_obj, temp_file_path)
-                pbar.update(temp_file_path.stat().st_size)
-                upload_path = str(temp_file_path)
-
-        return [token for token in [_upload_blob(upload_path, model_type)] if token]
+def upload_files(folder: str, model_type: str, quiet: bool = False) -> List[str]:  # noqa: FBT002, FBT001
```

Comment:
The return type of this function is different. You should also update `models.py` to reflect this and change the name.